src/ctap/timed_permission.rs

```diff
@@ -12,41 +12,52 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use libtock_drivers::timer::{ClockValue, Duration};
+use super::super::clock::{ClockInt, CtapInstant};
+use embedded_time::duration::Milliseconds;
 
-#[derive(Clone, Copy, Debug)]
+#[derive(Debug)]
 pub enum TimedPermission {
     Waiting,
-    Granted(ClockValue),
+    Granted(CtapInstant),
+}
+
+impl Clone for TimedPermission {
```

Comment:
It is not leftover from previous constraint. Without it, the compiler complains:
cannot move out of `self.hid.wink_permission` which is behind a mutable reference
  --> src/lib.rs:73:36
   |
73 |         self.hid.wink_permission = self.hid.wink_permission.check_expiration(now);
   |                                    ^^^^^^^^^^^^^^^^^^^^^^^^ move occurs because `self.hid.wink_permission` has type `TimedPermission`, which does not implement the `Copy` trait

Reply:
You don't have to manually implement Copy, though, try:
```
#[derive(Clone, Copy, Debug)]
pub enum TimedPermission {
```

Reply:
Could you try this? https://github.com/shilingwangggg/OpenSK/compare/feature/embedded_time...ia0:shiling_copy_clone?expand=1

It works for me. If it fails for you, could you provide me the command you use?

Reply:
It works with adding copy and clone to derive. @ia0 you're right it is a left over from the previous iteration with the generics.